virttest/test_setup.py

```diff
@@ -1438,6 +1438,38 @@ def check_vfs_count(self):
         expected_count = int((re.findall("(\d+)", self.driver_option)[0])) * len(self.get_pf_ids())
         return (self.get_vfs_count() == expected_count)
 
+    def set_linkvf_ib(self):
+        """
+        Enable the link for the VF's in SR-IOV Mellanox InfiniBand Controller
+        """
+        def port_id_vf():
+            return "11:22:33:44:55:66:%02x:00" % (
+                random.randint(0, 99),
+                )
+
+        pids = {}
+        cmd = "ls /sys/class/infiniband/mlx5_*/device/sriov/*"
+        dev = process.system_output(cmd, shell=True)
+        if not dev:
+            raise exceptions.TestSetupFail("No VF's found for set-up : %s"
+                                           % dev)
+
+        for line in dev.split('\n\n'):
+            key = line.split(':')[0]
+            value = {}
+            for attr in line.split(':')[1].split():
+                if attr == 'policy':
+                    value[attr] = 'Follow'
+                if attr in ('node', 'port'):
+                    value[attr] = port_id_vf()
+            pids[key] = value
+
+        for key, value in pids.iteritems():
+            for subkey, subvalue in pids[key].iteritems():
+                process.system("echo %s > %s" % (pids[key][subkey],
```

Comment:
1. it can be done in above for loop itself
2. we have to check the return value of process.system and return False on failure